src/microceph.py

```diff
@@ -42,7 +57,51 @@ def is_cluster_member(hostname: str) -> bool:
             raise e
 
 
-def add_osd_cmd(spec: str) -> None:
+def add_osd_cmd(spec: str, wal_dev: str = None, db_dev: str = None) -> None:
     """Executes MicroCeph add osd cmd with provided spec."""
     cmd = ["microceph", "disk", "add", spec]
+    if wal_dev:
+        cmd.extend(["--wal-device", wal_dev, "--wal-wipe"])
+    if db_dev:
+        cmd.extend(["--db-device", db_dev, "--db-wipe"])
     _run_cmd(cmd)
+
+
+def enroll_disks_as_osds(disks: list) -> None:
+    """Enrolls the provided block devices as OSDs."""
+    if not disks:
+        return
+
+    available_disks = []
+    for disk in disks:
+        if not _is_block_device_enrollable(disk):
+            err_str = f"provided disk {disk} is not enrollable as an OSD."
+            logger.error(err_str)
+            raise ValueError(err_str)
+
+        available_disks.append(disk)
+
+    # pass disks as space separated arguments.
+    add_osd_cmd(" ".join(available_disks))
+
+
+def _is_block_device_enrollable(disk) -> bool:
+    """Checks if the provided block device is enrollable as an OSD."""
+    device = json.loads(_run_cmd(["lsblk", f"{disk}", "--json"]).decode("utf-8"))["blockdevices"]
+
+    if len(device) == 0:
+        return False
+
+    if "loop" in device[0]["name"]:
```

Comment:
I realize this will catch all loopback devices, but it could also catch other things too. It would be better to use `device[0]["name"].startswith("/dev/loop")`, imo.